File: src/main/java/editor/database/symbol/ColorIntensity.java
package editor.database.symbol;

import editor.database.attributes.ManaType;

/**
 * This class represents a color intensity for a Symbol.  Only mana Symbols will have nonzero intensities
 * for any type of mana.  For any given mana Symbol, the sum of all of its color intensities should be
 * 1.
 *
 * @param color color to record intensity of
 * @param intensity intensity of the color
 * 
 * @author Alec Roelke
 */
public class ColorIntensity//(ManaType color, double intensity)
{
    public final ManaType color;
    public final double intensity;

    public ColorIntensity(ManaType c, double i)
    {
        color = c;
        intensity = i;
    }

    public ManaType color()
    {
        return color;
    }

    public double intensity()
    {
        return intensity;
    }
}

File: src/main/scala/editor/database/symbol/ColorIntensity.scala
package editor.database.symbol

import editor.database.attributes.ManaType

/**
 * An intensity for a mana type in a [[editor.database.symbol.Symbol]] or [[editor.database.attributes.ManaCost]].
 * 
 * @constructor create a new color intensity for a [[editor.database.attributes.ManaType]]
 * @param color type of mana
 * @param intensity intensity of that mana type
 * @note For any given mana [[editor.database.symbol.Symbol]], the sum of all of its components' intensities should be 1.
 * 
 * @author Alec Roelke
 */
case class ColorIntensity(color: ManaType, intensity: Double)